platforms/access-bridge/test/integration/access.ts

```diff
@@ -0,0 +1,179 @@
+import assert from 'assert';
+import { describe, test, before, after } from 'node:test';
+
+import { AccessController } from '../../src/controllers/access-controller.js';
+import { MockServer } from '../mock-server.js';
+import { ErrorCode } from '../../src/errors.js';
+import { ACCESS_TOKENS, AUTHORIZATION, ENDPOINTS, SITE_ID } from '../fixtures.js';
+import { MockAccessController } from '../mocks/access.js';
+
+describe('AccessController passport generate/refresh tests', async () => {
+  let mockServer: MockServer;
+  let accessController: AccessController;
+
+  before(async () => {
+    accessController = new MockAccessController();
+
+    const endpoints = {
+      [ENDPOINTS.GENERATE_TOKENS]: {
+        PUT: accessController.generatePassport,
+      },
+      [ENDPOINTS.REFRESH_TOKENS]: {
+        PUT: accessController.refreshPassport,
+      },
+    };
+
+    mockServer = await MockServer.create(endpoints);
+  });
+
+  await test('should generate passport access tokens without authorization', (t, done) => {
+    const requestOptions = {
+      method: 'PUT',
+      path: ENDPOINTS.GENERATE_TOKENS.replace(':site_id', SITE_ID.VALID),
+    };
+
+    mockServer
+      .request(requestOptions, (res) => {
+        assert.strictEqual(res.statusCode, 200);
+        let body = '';
+        res.on('data', (chunk) => {
+          body += chunk;
+        });
+        res.on('end', () => {
+          const responseBody = JSON.parse(body);
+          assert.strictEqual(responseBody.passport, ACCESS_TOKENS.PASSPORT.VALID);
+          assert.strictEqual(responseBody.refresh_token, ACCESS_TOKENS.REFRESH_TOKEN.VALID);
+          done();
+        });
+      })
+      .end();
+  });
+
+  await test('should generate passport access tokens with valid authorization', (t, done) => {
```

Comment:
Can this test and the 1 above be 2 test cases with different params but the same logic?

Reply:
https://mochajs.org/#dynamically-generating-tests